src/event.cpp

```diff
@@ -92,6 +92,9 @@ static owning_lock<event_handler_list_t> s_event_handlers;
 /// This is inspected by a signal handler. We assume no values in here overflow.
 static std::array<relaxed_atomic_t<uint32_t>, NSIG> s_observed_signals;
 
+/// List of events that have been sent but have not yet been delivered because they are blocked.
+static std::vector<std::shared_ptr<const event_t>> s_blocked_events{};
```

Comment:
we should probably make this an opaque rust type and move it back into `library_data_t` (to get rid of the global variable)